src/parser_data.h

```diff
@@ -187,53 +187,33 @@ struct ly_in;
 
 /** @} datavalidationoptions */
 
-/**
- * @ingroup datatree
- * @defgroup datavalidateop Operation to validate
- *
- * Operation provided to ::lyd_validate_op() to validate.
- *
- * The operation cannot be determined automatically since RPC/action and a reply to it share the common top level node
- * referencing the RPC/action schema node and may not have any input/output children to use for distinction.
- *
- * @{
- */
-typedef enum {
-    LYD_VALIDATE_OP_RPC = 1,   /**< Validate RPC/action request (input parameters). */
-    LYD_VALIDATE_OP_REPLY,     /**< Validate RPC/action reply (output parameters). */
-    LYD_VALIDATE_OP_NOTIF      /**< Validate Notification operation. */
-} LYD_VALIDATE_OP;
-
-/** @} datavalidateop */
-
 /**
  * @brief Parse (and validate) data from the input handler as a YANG data tree.
  *
  * @param[in] ctx Context to connect with the tree being built here.
+ * @param[in] parent Optional parent to connect the parsed nodes to.
  * @param[in] in The input handle to provide the dumped data in the specified @p format to parse (and validate).
  * @param[in] format Format of the input data to be parsed. Can be 0 to try to detect format from the input handler.
  * @param[in] parse_options Options for parser, see @ref dataparseroptions.
  * @param[in] validate_options Options for the validation phase, see @ref datavalidationoptions.
- * @param[out] tree Resulting data tree built from the input data. Note that NULL can be a valid result as a representation of an empty YANG data tree.
```

Comment:
Why aren't these notes valid now?

Reply:
I thought there was some useless information so I shortened it, maybe too much, I have added this note back.